File: ular-tangga-dalam-java/Floor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author USER
 */
public class Floor {
    int num;
    boolean empty;
    boolean snake;
    boolean ladder;

    public Floor(int num)
    {
        this.num = num;
        this.snake = false;
        this.ladder = false;
        this.empty = true;
    }

    public void setNum(int num) {
        this.num = num;
    }

    public void setSnake(boolean snake) {
        this.snake = snake;
    }

    public void setLadder(boolean ladder) {
        this.ladder = ladder;
    }

    public boolean isEmpty() {
        return empty;
    }

    public void setEmpty(boolean empty) {
        this.empty = empty;
    }
    
    public int getNum() {
        return num;
    }

    public boolean isSnake() {
        return this.snake;
    }

    public boolean isLadder() {
        return this.ladder;
    }
    
    /*public int getSnakeNum()
    {
        return this.snake.getNum();
    }
    
    public int getLadderNum()
    {
        return this.ladder.getNum();
    }*/
}

File: UlarTangga/Floor.rb
class Floor
  @num
  @snake
  @ladder
  
  def initialize(angka)
    @num =  angka
    @snake = false
    @ladder = false
  end
  
  def setNum(angka)
    @num =  angka
  end
  
  def getNum
    return @num
  end
  
  def setSnake(ular)
    @snake = ular
  end
  
  def setLadder(tangga)
    @ladder = tangga
  end
  
  def isSnake
    return @snake
  end
  
  def isLadder
    return @ladder
  end
end
